File: solutions2020/PRB6/bla269@fuhsd.student.org.cpp
#include <iostream>
using namespace std;
long long N, K,current = 0;
int A[1000000];
bool ansFound = true;
int main(){
	int numCases;
	cin >>numCases;
	for (int cases= 0; cases < numCases; cases++){
		cin >> N >> K;
		current = 0;
		for (int i = 0; i < N; i++){
			cin >> A[i];
		}
		
		for (int i = 0; i < N; i++){
			ansFound = true;
			for (int j = 1; j <= K; j++){
				if (current + j == N){
					ansFound = true;
					break;
				}
				else if (A[current]<A[current+j]){
					current = current + j;
					i = current - 1;
					ansFound = false;
					break;
				}
			}
			if(ansFound){
				cout << A[current]<<endl;
				break;
			}
		}
	}
}

File: solutions2020/PRA5/bla269@fuhsd.student.org.java
#include <iostream>
#include <string>
char vowels[5] = {'a','e','i','o','u'};
int numConsonants, continous;
using namespace std;
bool isVowel(char x){
	bool vowel = false;
	for (int i = 0; i < 5; i++){
		if (x == vowels[i]){
			vowel = true;
		}
	}
	return vowel;
}
int main(){
	int numCases;
	string str;

	cin >>numCases;
	for (int cases = 0; cases < numCases; cases++){
		cin >> str;
		numConsonants = 0;
		continous = 1;
		for (int i = 0; i < str.length(); i++){
			if (isVowel(str[i])){
				numConsonants= 0;
				continous = 1;
			}
			else{
				numConsonants++;
			} 
			if (str[i] == str[i-1]){
				continous++;
			}
			else{
				continous = 1;
			}
			if (numConsonants >=3 && continous <3){
				str.insert(i," ");
			}
			
		}
		cout << str<<endl;
	}
}
